tests/components/blue_current/__init__.py

```diff
@@ -19,35 +19,21 @@
 }
 
 
-async def connect(client, token: str):
-    """Set the connected event."""
-    if client.connect_exception:
-        # Raise exception once.
-        temp = client.connect_exception
-        client.connect_exception = None
-        raise temp
-
-    client.connected.set()
-    client.connected.clear()
-
-
-async def start_loop(client, receiver):
+async def start_loop(hass: HomeAssistant, client, receiver):
```

Comment:
I'd consider nesting this function inside the `create_client` function so we don't need to pass parameters that are not really part of the client method signature.

We can then create the future outside this function.